mne/tests/test_evoked.py

```diff
@@ -759,3 +761,24 @@ def test_hilbert():
 
 
 run_tests_if_main()
+
+
+def fun(evoked_data):
+    """Auxiliary function to test apply_function."""
+    mult_array = np.random.rand(1000)
+    product = np.multiply(mult_array, evoked_data)
+    return np.asarray(product)
+
+
+def test_apply_function_evk():
+    """Check the apply_function method for evoked data."""
+    # create fake evoked data to use for checking apply_function
+    data = np.random.rand(2, 10, 1000)
+    events = np.array([[0, 0, 1], [INT32_MAX, 0, 2]])
+    info = mne.create_info(10, 1000., 'eeg')
+    epochs = mne.EpochsArray(data, info, events)
+    evoked = epochs.average(method='mean')
+    evoked_data = evoked.data
+    # check apply_function channel-wise
+    applied = evoked.apply_function(fun)
+    assert np.shape(applied.data) == np.shape(evoked_data)
```

Comment:
can we test something more stringent besides just the data shape?  (like that the data values end up how we expect them to end up)?